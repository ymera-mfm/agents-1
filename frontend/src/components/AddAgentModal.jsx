import React, { useState } from 'react';
import { X, Zap, Code, Camera, Database, Shield } from 'lucide-react';

const agentTypes = [
  { id: 'code-analyzer', label: 'Code Analyzer', icon: Code, color: '#00f5ff' },
  { id: 'ui-designer', label: 'UI Designer', icon: Camera, color: '#ff00aa' },
  { id: 'backend-dev', label: 'Backend Dev', icon: Database, color: '#00ff88' },
  { id: 'security', label: 'Security', icon: Shield, color: '#ffaa00' },
  { id: 'optimizer', label: 'Optimizer', icon: Zap, color: '#aa00ff' },
];

export const AddAgentModal = ({ onClose, onAdd }) => {
  const [name, setName] = useState('');
  const [type, setType] = useState('code-analyzer');
  const [status, setStatus] = useState('idle');
  const [description, setDescription] = useState('');

  const handleSubmit = () => {
    const agentType = agentTypes.find((t) => t.id === type);
    const newAgent = {
      id: `agent-${Date.now()}`,
      name,
      type,
      status,
      description: description || `${name} is a specialized ${type.replace('-', ' ')}.`,
      tasksCompleted: 0,
      efficiency: 95,
      icon: agentType.icon.name,
      color: agentType.color,
    };
    onAdd(newAgent);
    onClose();
  };

  return (
    <div className="fixed inset-0 flex items-center justify-center bg-black/50 backdrop-blur-sm z-50 p-4">
      <div
        className="w-full max-w-md rounded-2xl overflow-hidden"
        style={{
          background: 'rgba(15, 15, 15, 0.98)',
          backdropFilter: 'blur(40px)',
          border: '1px solid rgba(255, 255, 255, 0.1)',
        }}
      >
        <div className="p-6 border-b border-gray-800">
          <div className="flex items-center justify-between mb-4">
            <h2 className="text-2xl font-bold text-white">Add New Agent</h2>
            <button
              onClick={onClose}
              className="p-2 rounded-lg hover:bg-white/10 transition-colors"
            >
              <X size={20} className="text-gray-400" />
            </button>
          </div>
        </div>
        <div className="p-6 space-y-4">
          <div>
            <label className="block text-sm text-gray-400 mb-2">Agent Name</label>
            <input
              type="text"
              value={name}
              onChange={(e) => setName(e.target.value)}
              placeholder="e.g., CodeAnalyzer Alpha"
              className="w-full px-4 py-3 rounded-xl bg-white/5 border border-white/10 text-white placeholder-gray-500 focus:border-cyan-400 focus:outline-none transition-colors"
            />
          </div>
          <div>
            <label className="block text-sm text-gray-400 mb-2">Agent Type</label>
            <div className="grid grid-cols-3 gap-3">
              {agentTypes.map((agentType) => (
                <button
                  key={agentType.id}
                  onClick={() => setType(agentType.id)}
                  className={`flex flex-col items-center p-3 rounded-xl border-2 transition-all ${
                    type === agentType.id
                      ? 'border-cyan-400 bg-cyan-500/10'
                      : 'border-white/10 bg-white/5 hover:border-cyan-400/30'
                  }`}
                >
                  <agentType.icon size={24} style={{ color: agentType.color }} />
                  <span className="text-xs mt-2 text-gray-300">{agentType.label}</span>
                </button>
              ))}
            </div>
          </div>
          <div>
            <label className="block text-sm text-gray-400 mb-2">Initial Status</label>
            <select
              value={status}
              onChange={(e) => setStatus(e.target.value)}
              className="w-full px-4 py-3 rounded-xl bg-white/5 border border-white/10 text-white focus:border-cyan-400 focus:outline-none transition-colors"
            >
              <option value="idle">Idle</option>
              <option value="thinking">Thinking</option>
              <option value="working">Working</option>
            </select>
          </div>
          <div>
            <label className="block text-sm text-gray-400 mb-2">Description (Optional)</label>
            <textarea
              value={description}
              onChange={(e) => setDescription(e.target.value)}
              placeholder="Describe the agent's specialty..."
              rows="3"
              className="w-full px-4 py-3 rounded-xl bg-white/5 border border-white/10 text-white placeholder-gray-500 focus:border-cyan-400 focus:outline-none transition-colors resize-none"
            />
          </div>
          <div className="flex justify-end gap-3 pt-4">
            <button
              onClick={onClose}
              className="px-6 py-2 rounded-xl bg-white/5 hover:bg-white/10 border border-white/10 text-white transition-colors"
            >
              Cancel
            </button>
            <button
              onClick={handleSubmit}
              disabled={!name.trim()}
              className="px-6 py-2 rounded-xl bg-cyan-500 hover:bg-cyan-600 text-white font-medium transition-colors disabled:opacity-50"
            >
              Add Agent
            </button>
          </div>
        </div>
      </div>
    </div>
  );
};
