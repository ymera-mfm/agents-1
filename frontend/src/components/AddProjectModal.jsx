import React, { useState } from 'react';
import { X, Plus } from 'lucide-react';

export const AddProjectModal = ({ onClose, onAdd }) => {
  const [name, setName] = useState('');
  const [description, setDescription] = useState('');
  const [status, setStatus] = useState('planning');
  const [startDate, setStartDate] = useState(new Date().toISOString().split('T')[0]);
  const [endDate, setEndDate] = useState(
    new Date(Date.now() + 30 * 24 * 60 * 60 * 1000).toISOString().split('T')[0]
  );
  const [assignedAgents, setAssignedAgents] = useState([]);

  const handleSubmit = () => {
    const newProject = {
      name,
      description,
      status,
      startDate,
      estimatedCompletion: endDate,
      progress: 0,
      assignedAgents,
    };
    onAdd(newProject);
    onClose();
  };

  return (
    <div className="fixed inset-0 flex items-center justify-center bg-black/50 backdrop-blur-sm z-50 p-4">
      <div
        className="w-full max-w-2xl rounded-2xl overflow-hidden"
        style={{
          background: 'rgba(15, 15, 15, 0.98)',
          backdropFilter: 'blur(40px)',
          border: '1px solid rgba(255, 255, 255, 0.1)',
        }}
      >
        <div className="p-6 border-b border-gray-800">
          <div className="flex items-center justify-between mb-4">
            <h2 className="text-2xl font-bold text-white">Create New Project</h2>
            <button
              onClick={onClose}
              className="p-2 rounded-lg hover:bg-white/10 transition-colors"
              aria-label="Close"
            >
              <X size={20} className="text-gray-400" />
            </button>
          </div>
        </div>
        <div className="p-6 space-y-6">
          <div className="grid grid-cols-1 md:grid-cols-2 gap-6">
            <div>
              <label className="block text-sm text-gray-400 mb-2">Project Name</label>
              <input
                type="text"
                value={name}
                onChange={(e) => setName(e.target.value)}
                placeholder="e.g., E-Commerce Platform"
                className="w-full px-4 py-3 rounded-xl bg-white/5 border border-white/10 text-white placeholder-gray-500 focus:border-cyan-400 focus:outline-none transition-colors"
              />
            </div>
            <div>
              <label className="block text-sm text-gray-400 mb-2">Status</label>
              <select
                value={status}
                onChange={(e) => setStatus(e.target.value)}
                className="w-full px-4 py-3 rounded-xl bg-white/5 border border-white/10 text-white focus:border-cyan-400 focus:outline-none transition-colors"
              >
                <option value="planning">Planning</option>
                <option value="in_progress">In Progress</option>
                <option value="on_hold">On Hold</option>
              </select>
            </div>
          </div>

          <div>
            <label className="block text-sm text-gray-400 mb-2">Description</label>
            <textarea
              value={description}
              onChange={(e) => setDescription(e.target.value)}
              placeholder="Describe the project goals and scope..."
              rows="4"
              className="w-full px-4 py-3 rounded-xl bg-white/5 border border-white/10 text-white placeholder-gray-500 focus:border-cyan-400 focus:outline-none transition-colors resize-none"
            />
          </div>

          <div className="grid grid-cols-1 md:grid-cols-2 gap-6">
            <div>
              <label className="block text-sm text-gray-400 mb-2">Start Date</label>
              <input
                type="date"
                value={startDate}
                onChange={(e) => setStartDate(e.target.value)}
                className="w-full px-4 py-3 rounded-xl bg-white/5 border border-white/10 text-white focus:border-cyan-400 focus:outline-none transition-colors"
              />
            </div>
            <div>
              <label className="block text-sm text-gray-400 mb-2">Estimated Completion</label>
              <input
                type="date"
                value={endDate}
                onChange={(e) => setEndDate(e.target.value)}
                className="w-full px-4 py-3 rounded-xl bg-white/5 border border-white/10 text-white focus:border-cyan-400 focus:outline-none transition-colors"
              />
            </div>
          </div>

          <div>
            <label className="block text-sm text-gray-400 mb-2">Assigned Agents</label>
            <div className="flex flex-wrap gap-2 mb-3">
              {assignedAgents.map((agentId) => (
                <div
                  key={agentId}
                  className="flex items-center gap-2 px-3 py-1 rounded-full bg-cyan-500/20 border border-cyan-500/30 text-white"
                >
                  <span>{agentId}</span>
                  <button
                    onClick={() => setAssignedAgents(assignedAgents.filter((id) => id !== agentId))}
                    className="p-0.5 rounded-full hover:bg-white/10"
                  >
                    <X size={14} />
                  </button>
                </div>
              ))}
            </div>
            <div className="flex gap-2">
              <select
                value=""
                onChange={(e) => {
                  if (e.target.value && !assignedAgents.includes(e.target.value)) {
                    setAssignedAgents([...assignedAgents, e.target.value]);
                  }
                }}
                className="flex-1 px-4 py-2 rounded-lg bg-white/5 border border-white/10 text-white"
              >
                <option value="">Select an agent</option>
                <option value="agent-001">CodeAnalyzer Alpha</option>
                <option value="agent-002">UIDesigner Beta</option>
                <option value="agent-003">BackendDev Gamma</option>
              </select>
              <button className="px-4 py-2 rounded-lg bg-cyan-500 hover:bg-cyan-600 text-white transition-colors">
                <Plus size={18} />
              </button>
            </div>
          </div>

          <div className="flex justify-end gap-3 pt-6 border-t border-gray-800">
            <button
              onClick={onClose}
              className="px-6 py-2 rounded-xl bg-white/5 hover:bg-white/10 border border-white/10 text-white transition-colors"
            >
              Cancel
            </button>
            <button
              onClick={handleSubmit}
              disabled={!name.trim()}
              className="px-6 py-2 rounded-xl bg-cyan-500 hover:bg-cyan-600 text-white font-medium transition-colors disabled:opacity-50"
            >
              Create Project
            </button>
          </div>
        </div>
      </div>
    </div>
  );
};
