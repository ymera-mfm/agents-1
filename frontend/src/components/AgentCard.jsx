import React from 'react';
import { StatusBadge } from './StatusBadge';
import { Code, Camera, Database, Shield, Zap } from 'lucide-react';

const iconMap = {
  Code: Code,
  Camera: Camera,
  Database: Database,
  Shield: Shield,
  Zap: Zap,
};

export const AgentCard = React.memo(({ agent, onClick }) => {
  const Icon = iconMap[agent.icon] || Zap;

  return (
    <div
      onClick={() => onClick(agent)}
      className="group relative overflow-hidden rounded-2xl cursor-pointer transition-all duration-300 hover:scale-105"
      style={{
        background: 'rgba(20, 20, 20, 0.9)',
        backdropFilter: 'blur(25px)',
        border: '1px solid rgba(255, 255, 255, 0.1)',
        boxShadow: '0 8px 32px rgba(0, 0, 0, 0.37)',
      }}
      aria-label={`View ${agent.name} details`}
      role="button"
      tabIndex="0"
      onKeyDown={(e) => e.key === 'Enter' && onClick(agent)}
    >
      <div
        className="absolute inset-0 opacity-0 group-hover:opacity-20 transition-opacity duration-300"
        style={{
          background: `linear-gradient(135deg, ${agent.color}40, transparent)`,
        }}
      />
      <div className="relative p-6 space-y-4">
        <div className="flex items-start justify-between">
          <div
            className="p-3 rounded-xl transition-transform group-hover:scale-110"
            style={{
              background: `${agent.color}20`,
              border: `1px solid ${agent.color}40`,
            }}
          >
            <Icon size={24} style={{ color: agent.color }} />
          </div>
          <StatusBadge status={agent.status} />
        </div>
        <div>
          <h3 className="text-xl font-bold text-white mb-1">{agent.name}</h3>
          <p className="text-sm text-gray-400 font-mono">{agent.type.toUpperCase()}</p>
        </div>
        <p className="text-sm text-gray-500 leading-relaxed">{agent.description}</p>
        <div className="flex items-center justify-between pt-4 border-t border-gray-800">
          <div>
            <div className="text-2xl font-bold" style={{ color: agent.color }}>
              {agent.tasksCompleted}
            </div>
            <div className="text-xs text-gray-500">Tasks</div>
          </div>
          <div>
            <div className="text-2xl font-bold" style={{ color: agent.color }}>
              {agent.efficiency}%
            </div>
            <div className="text-xs text-gray-500">Efficiency</div>
          </div>
        </div>
      </div>
    </div>
  );
});
