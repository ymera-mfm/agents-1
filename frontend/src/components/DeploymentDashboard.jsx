// components/DeploymentDashboard.jsx
import React, { useState } from 'react';
import { motion, AnimatePresence } from 'framer-motion';
import { Play, Square, RefreshCw, CheckCircle, XCircle, Clock, Settings } from 'lucide-react';

export const DeploymentDashboard = () => {
  const [deployments, setDeployments] = useState([]);
  const [isDeploying, setIsDeploying] = useState(false);
  const [deploymentProgress, setDeploymentProgress] = useState(0);

  const deploymentStages = [
    { name: 'Code Compilation', duration: 30 },
    { name: 'Dependency Installation', duration: 45 },
    { name: 'Testing', duration: 60 },
    { name: 'Asset Optimization', duration: 30 },
    { name: 'Deployment', duration: 45 },
  ];

  const startDeployment = async () => {
    setIsDeploying(true);
    setDeploymentProgress(0);

    const deployment = {
      id: Date.now(),
      version: `v1.${deployments.length + 1}.0`,
      startTime: new Date(),
      status: 'running',
      stages: deploymentStages.map((stage) => ({ ...stage, status: 'pending' })),
    };

    setDeployments((prev) => [deployment, ...prev]);

    // Simulate deployment process
    for (let i = 0; i < deploymentStages.length; i++) {
      await new Promise((resolve) => setTimeout(resolve, deploymentStages[i].duration * 10));

      setDeployments((prev) =>
        prev.map((dep) =>
          dep.id === deployment.id
            ? {
                ...dep,
                stages: dep.stages.map((stage, index) =>
                  index === i ? { ...stage, status: 'completed' } : stage
                ),
              }
            : dep
        )
      );

      setDeploymentProgress(((i + 1) / deploymentStages.length) * 100);
    }

    // Finalize deployment
    setDeployments((prev) =>
      prev.map((dep) =>
        dep.id === deployment.id
          ? {
              ...dep,
              status: 'completed',
              endTime: new Date(),
              success: true,
            }
          : dep
      )
    );

    setIsDeploying(false);
    setDeploymentProgress(0);
  };

  const cancelDeployment = () => {
    setIsDeploying(false);
    setDeploymentProgress(0);

    setDeployments((prev) =>
      prev.map((dep) =>
        dep.status === 'running' ? { ...dep, status: 'cancelled', endTime: new Date() } : dep
      )
    );
  };

  const DeploymentStage = ({ stage }) => (
    <div className="flex items-center space-x-3 p-3 bg-white/5 rounded-lg">
      <div
        className={`w-8 h-8 rounded-full flex items-center justify-center ${
          stage.status === 'completed'
            ? 'bg-green-500/20 text-green-400'
            : stage.status === 'running'
              ? 'bg-cyan-500/20 text-cyan-400 animate-pulse'
              : 'bg-gray-500/20 text-gray-400'
        }`}
      >
        {stage.status === 'completed' ? (
          <CheckCircle className="w-4 h-4" />
        ) : stage.status === 'running' ? (
          <RefreshCw className="w-4 h-4" />
        ) : (
          <Clock className="w-4 h-4" />
        )}
      </div>

      <div className="flex-1">
        <div className="text-white font-medium">{stage.name}</div>
        <div className="text-sm text-gray-400">{stage.duration}s estimated</div>
      </div>

      {stage.status === 'running' && (
        <div className="w-20 bg-gray-700 rounded-full h-2">
          <motion.div
            className="h-2 rounded-full bg-cyan-500"
            initial={{ width: 0 }}
            animate={{ width: '100%' }}
            transition={{ duration: stage.duration / 10, ease: 'linear' }}
          />
        </div>
      )}
    </div>
  );

  return (
    <div className="space-y-6">
      <div className="flex items-center justify-between">
        <div>
          <h3 className="text-2xl font-bold text-white">Deployment Center</h3>
          <p className="text-gray-400">Manage application deployments and CI/CD pipelines</p>
        </div>

        <div className="flex items-center space-x-3">
          <button
            onClick={startDeployment}
            disabled={isDeploying}
            className="flex items-center space-x-2 px-4 py-2 bg-gradient-to-r from-green-500 to-emerald-600 text-white rounded-lg hover:from-green-600 hover:to-emerald-700 transition disabled:opacity-50"
          >
            <Play className="w-4 h-4" />
            <span>{isDeploying ? 'Deploying...' : 'Deploy Now'}</span>
          </button>

          {isDeploying && (
            <button
              onClick={cancelDeployment}
              className="flex items-center space-x-2 px-4 py-2 bg-red-500/20 text-red-400 rounded-lg hover:bg-red-500/30 transition"
            >
              <Square className="w-4 h-4" />
              <span>Cancel</span>
            </button>
          )}
        </div>
      </div>

      {/* Current Deployment Progress */}
      {isDeploying && (
        <motion.div
          initial={{ opacity: 0, height: 0 }}
          animate={{ opacity: 1, height: 'auto' }}
          className="backdrop-blur-xl bg-white/5 border border-white/10 rounded-xl p-6"
        >
          <div className="flex items-center justify-between mb-4">
            <h4 className="text-lg font-semibold text-white">Deployment in Progress</h4>
            <div className="text-cyan-400 font-semibold">{deploymentProgress.toFixed(0)}%</div>
          </div>

          <div className="w-full bg-gray-700 rounded-full h-3 mb-6">
            <motion.div
              className="h-3 rounded-full bg-gradient-to-r from-cyan-500 to-blue-600"
              initial={{ width: 0 }}
              animate={{ width: `${deploymentProgress}%` }}
              transition={{ duration: 0.5 }}
            />
          </div>

          <div className="space-y-3">
            {deployments[0]?.stages.map((stage, index) => (
              <DeploymentStage
                key={index}
                stage={stage}
                index={index}
                currentStage={deploymentProgress / (100 / deploymentStages.length)}
              />
            ))}
          </div>
        </motion.div>
      )}

      {/* Deployment History */}
      <div className="backdrop-blur-xl bg-white/5 border border-white/10 rounded-xl p-6">
        <h4 className="text-lg font-semibold text-white mb-4">Deployment History</h4>

        <div className="space-y-3">
          <AnimatePresence>
            {deployments.map((deployment) => (
              <motion.div
                key={deployment.id}
                initial={{ opacity: 0, x: -20 }}
                animate={{ opacity: 1, x: 0 }}
                exit={{ opacity: 0, x: 20 }}
                className="flex items-center justify-between p-4 bg-white/5 rounded-lg"
              >
                <div className="flex items-center space-x-4">
                  <div
                    className={`w-12 h-12 rounded-lg flex items-center justify-center ${
                      deployment.status === 'completed'
                        ? 'bg-green-500/20'
                        : deployment.status === 'running'
                          ? 'bg-cyan-500/20 animate-pulse'
                          : 'bg-red-500/20'
                    }`}
                  >
                    {deployment.status === 'completed' ? (
                      <CheckCircle className="w-6 h-6 text-green-400" />
                    ) : deployment.status === 'running' ? (
                      <RefreshCw className="w-6 h-6 text-cyan-400" />
                    ) : (
                      <XCircle className="w-6 h-6 text-red-400" />
                    )}
                  </div>

                  <div>
                    <div className="text-white font-semibold">{deployment.version}</div>
                    <div className="text-sm text-gray-400">
                      {deployment.startTime.toLocaleString()}
                      {deployment.endTime && ` - ${deployment.endTime.toLocaleTimeString()}`}
                    </div>
                  </div>
                </div>

                <div className="text-right">
                  <div
                    className={`text-sm font-semibold ${
                      deployment.status === 'completed'
                        ? 'text-green-400'
                        : deployment.status === 'running'
                          ? 'text-cyan-400'
                          : 'text-red-400'
                    }`}
                  >
                    {deployment.status.charAt(0).toUpperCase() + deployment.status.slice(1)}
                  </div>
                  <div className="text-xs text-gray-400">
                    {deployment.stages.filter((s) => s.status === 'completed').length} /{' '}
                    {deployment.stages.length} stages
                  </div>
                </div>
              </motion.div>
            ))}
          </AnimatePresence>

          {deployments.length === 0 && (
            <div className="text-center py-8 text-gray-400">
              <Settings className="w-12 h-12 mx-auto mb-3 opacity-50" />
              <div>No deployments yet</div>
              <div className="text-sm">Start your first deployment to see history</div>
            </div>
          )}
        </div>
      </div>

      {/* Environment Configuration */}
      <div className="grid grid-cols-1 lg:grid-cols-2 gap-6">
        <div className="backdrop-blur-xl bg-white/5 border border-white/10 rounded-xl p-6">
          <h4 className="text-lg font-semibold text-white mb-4">Environment Settings</h4>

          <div className="space-y-4">
            <div>
              <label className="block text-sm font-medium text-gray-300 mb-2">Environment</label>
              <select className="w-full px-4 py-2 bg-white/5 border border-white/10 rounded-lg text-white focus:outline-none focus:ring-2 focus:ring-cyan-500">
                <option>Production</option>
                <option>Staging</option>
                <option>Development</option>
              </select>
            </div>

            <div>
              <label className="block text-sm font-medium text-gray-300 mb-2">Build Command</label>
              <input
                type="text"
                defaultValue="npm run build"
                className="w-full px-4 py-2 bg-white/5 border border-white/10 rounded-lg text-white focus:outline-none focus:ring-2 focus:ring-cyan-500"
              />
            </div>

            <div>
              <label className="block text-sm font-medium text-gray-300 mb-2">Node Version</label>
              <select className="w-full px-4 py-2 bg-white/5 border border-white/10 rounded-lg text-white focus:outline-none focus:ring-2 focus:ring-cyan-500">
                <option>18.x</option>
                <option>20.x</option>
                <option>16.x</option>
              </select>
            </div>
          </div>
        </div>

        <div className="backdrop-blur-xl bg-white/5 border border-white/10 rounded-xl p-6">
          <h4 className="text-lg font-semibold text-white mb-4">Deployment Statistics</h4>

          <div className="space-y-4">
            <div className="flex justify-between items-center p-3 bg-white/5 rounded-lg">
              <span className="text-gray-400">Total Deployments</span>
              <span className="text-white font-semibold">{deployments.length}</span>
            </div>

            <div className="flex justify-between items-center p-3 bg-white/5 rounded-lg">
              <span className="text-gray-400">Success Rate</span>
              <span className="text-green-400 font-semibold">
                {deployments.filter((d) => d.status === 'completed').length > 0
                  ? `${((deployments.filter((d) => d.status === 'completed').length / deployments.length) * 100).toFixed(0)}%`
                  : '0%'}
              </span>
            </div>

            <div className="flex justify-between items-center p-3 bg-white/5 rounded-lg">
              <span className="text-gray-400">Average Duration</span>
              <span className="text-cyan-400 font-semibold">
                {deploymentStages.reduce((sum, stage) => sum + stage.duration, 0)}s
              </span>
            </div>

            <div className="flex justify-between items-center p-3 bg-white/5 rounded-lg">
              <span className="text-gray-400">Last Deployment</span>
              <span className="text-white font-semibold">{deployments[0]?.version || 'None'}</span>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};
