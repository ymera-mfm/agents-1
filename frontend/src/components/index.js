// Auto-generated component exports
// This file makes imports easier: import { Component } from './components'
