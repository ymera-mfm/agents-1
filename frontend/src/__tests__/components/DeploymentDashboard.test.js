import React from 'react';
import { render } from '@testing-library/react';
import { DeploymentDashboard } from '../../components/DeploymentDashboard';

describe('DeploymentDashboard', () => {
  it('renders without crashing', () => {
    const { container } = render(<DeploymentDashboard />);
    expect(container).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { container } = render(<DeploymentDashboard />);
    expect(container).toMatchSnapshot();
  });

  // Add more specific tests for DeploymentDashboard
});
