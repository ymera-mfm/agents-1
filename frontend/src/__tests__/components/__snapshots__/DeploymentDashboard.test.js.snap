// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeploymentDashboard matches snapshot 1`] = `
<div>
  <div
    class="space-y-6"
  >
    <div
      class="flex items-center justify-between"
    >
      <div>
        <h3
          class="text-2xl font-bold text-white"
        >
          Deployment Center
        </h3>
        <p
          class="text-gray-400"
        >
          Manage application deployments and CI/CD pipelines
        </p>
      </div>
      <div
        class="flex items-center space-x-3"
      >
        <button
          class="flex items-center space-x-2 px-4 py-2 bg-gradient-to-r from-green-500 to-emerald-600 text-white rounded-lg hover:from-green-600 hover:to-emerald-700 transition disabled:opacity-50"
        >
          <svg
            class="w-4 h-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polygon
              points="5 3 19 12 5 21 5 3"
            />
          </svg>
          <span>
            Deploy Now
          </span>
        </button>
      </div>
    </div>
    <div
      class="backdrop-blur-xl bg-white/5 border border-white/10 rounded-xl p-6"
    >
      <h4
        class="text-lg font-semibold text-white mb-4"
      >
        Deployment History
      </h4>
      <div
        class="space-y-3"
      >
        <div
          class="text-center py-8 text-gray-400"
        >
          <svg
            class="w-12 h-12 mx-auto mb-3 opacity-50"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12.22 2h-.44a2 2 0 0 0-2 2v.18a2 2 0 0 1-1 1.73l-.43.25a2 2 0 0 1-2 0l-.15-.08a2 2 0 0 0-2.73.73l-.22.38a2 2 0 0 0 .73 2.73l.15.1a2 2 0 0 1 1 1.72v.51a2 2 0 0 1-1 1.74l-.15.09a2 2 0 0 0-.73 2.73l.22.38a2 2 0 0 0 2.73.73l.15-.08a2 2 0 0 1 2 0l.43.25a2 2 0 0 1 1 1.73V20a2 2 0 0 0 2 2h.44a2 2 0 0 0 2-2v-.18a2 2 0 0 1 1-1.73l.43-.25a2 2 0 0 1 2 0l.15.08a2 2 0 0 0 2.73-.73l.22-.39a2 2 0 0 0-.73-2.73l-.15-.08a2 2 0 0 1-1-1.74v-.5a2 2 0 0 1 1-1.74l.15-.09a2 2 0 0 0 .73-2.73l-.22-.38a2 2 0 0 0-2.73-.73l-.15.08a2 2 0 0 1-2 0l-.43-.25a2 2 0 0 1-1-1.73V4a2 2 0 0 0-2-2z"
            />
            <circle
              cx="12"
              cy="12"
              r="3"
            />
          </svg>
          <div>
            No deployments yet
          </div>
          <div
            class="text-sm"
          >
            Start your first deployment to see history
          </div>
        </div>
      </div>
    </div>
    <div
      class="grid grid-cols-1 lg:grid-cols-2 gap-6"
    >
      <div
        class="backdrop-blur-xl bg-white/5 border border-white/10 rounded-xl p-6"
      >
        <h4
          class="text-lg font-semibold text-white mb-4"
        >
          Environment Settings
        </h4>
        <div
          class="space-y-4"
        >
          <div>
            <label
              class="block text-sm font-medium text-gray-300 mb-2"
            >
              Environment
            </label>
            <select
              class="w-full px-4 py-2 bg-white/5 border border-white/10 rounded-lg text-white focus:outline-none focus:ring-2 focus:ring-cyan-500"
            >
              <option>
                Production
              </option>
              <option>
                Staging
              </option>
              <option>
                Development
              </option>
            </select>
          </div>
          <div>
            <label
              class="block text-sm font-medium text-gray-300 mb-2"
            >
              Build Command
            </label>
            <input
              class="w-full px-4 py-2 bg-white/5 border border-white/10 rounded-lg text-white focus:outline-none focus:ring-2 focus:ring-cyan-500"
              type="text"
              value="npm run build"
            />
          </div>
          <div>
            <label
              class="block text-sm font-medium text-gray-300 mb-2"
            >
              Node Version
            </label>
            <select
              class="w-full px-4 py-2 bg-white/5 border border-white/10 rounded-lg text-white focus:outline-none focus:ring-2 focus:ring-cyan-500"
            >
              <option>
                18.x
              </option>
              <option>
                20.x
              </option>
              <option>
                16.x
              </option>
            </select>
          </div>
        </div>
      </div>
      <div
        class="backdrop-blur-xl bg-white/5 border border-white/10 rounded-xl p-6"
      >
        <h4
          class="text-lg font-semibold text-white mb-4"
        >
          Deployment Statistics
        </h4>
        <div
          class="space-y-4"
        >
          <div
            class="flex justify-between items-center p-3 bg-white/5 rounded-lg"
          >
            <span
              class="text-gray-400"
            >
              Total Deployments
            </span>
            <span
              class="text-white font-semibold"
            >
              0
            </span>
          </div>
          <div
            class="flex justify-between items-center p-3 bg-white/5 rounded-lg"
          >
            <span
              class="text-gray-400"
            >
              Success Rate
            </span>
            <span
              class="text-green-400 font-semibold"
            >
              0%
            </span>
          </div>
          <div
            class="flex justify-between items-center p-3 bg-white/5 rounded-lg"
          >
            <span
              class="text-gray-400"
            >
              Average Duration
            </span>
            <span
              class="text-cyan-400 font-semibold"
            >
              210
              s
            </span>
          </div>
          <div
            class="flex justify-between items-center p-3 bg-white/5 rounded-lg"
          >
            <span
              class="text-gray-400"
            >
              Last Deployment
            </span>
            <span
              class="text-white font-semibold"
            >
              None
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
